src/jv_aux.c

```diff
@@ -38,29 +46,33 @@ static int parse_slice(jv j, jv slice, int* pstart, int* pend) {
       jv_get_kind(end_jv) != JV_KIND_NUMBER) {
     jv_free(start_jv);
     jv_free(end_jv);
-    return 0;
-  } else {
-    double dstart = jv_number_value(start_jv);
-    double dend = jv_number_value(end_jv);
-    jv_free(start_jv);
-    jv_free(end_jv);
-    if (dstart < 0) dstart += len;
-    if (dend < 0) dend += len;
-    if (dstart < 0) dstart = 0;
-    if (dstart > len) dstart = len;
-
-    int start = (int)dstart;
-    int end = (dend > len) ? len : (int)dend;
-    // Ends are exclusive but e.g. 1 < 1.5 so :1.5 should be :2 not :1
-    if(end < dend) end += 1;
-
-    if (end > len) end = len;
-    if (end < start) end = start;
-    assert(0 <= start && start <= end && end <= len);
-    *pstart = start;
-    *pend = end;
-    return 1;
-  }
+    return jv_invalid_with_msg(jv_string("Array/string slice indices must be integers"));
+  }
+
+  double dstart = jv_number_value(start_jv);
+  double dend = jv_number_value(end_jv);
+  int start, end;
+
+  jv_free(start_jv);
+  jv_free(end_jv);
+  if (isnan(dstart)) dstart = 0;
+  if (dstart < 0)    dstart += len;
+  if (dstart < 0)    dstart = 0;
+  if (dstart > len)  dstart = len;
+  start = dstart > INT_MAX ? INT_MAX : (int)dstart; // Rounds down
+
+  if (isnan(dend))   dend = len;
+  if (dend < 0)       dend += len;
+  end = dend > INT_MAX ? INT_MAX : (int)dend;
```

Comment:
I'm afraid when `dend < (double)INT_MIN - len` integer casting is an undefined behavior.